pkg/controller/metrics/metrics.go

```diff
@@ -595,5 +595,17 @@ func GetClusterDeploymentType(obj metav1.Object) string {
 func ObserveControllerReconcileTime(controller string, start time.Time, outcome ReconcileOutcome, logger log.FieldLogger) {
 	dur := time.Since(start)
 	metricControllerReconcileTime.WithLabelValues(controller, string(outcome)).Observe(dur.Seconds())
-	logger.WithFields(log.Fields{"elapsedMillis": dur.Milliseconds(), "outcome": outcome}).Info("reconcile complete")
+	fields := log.Fields{"elapsedMillis": dur.Milliseconds(), "outcome": outcome}
+	// Add a log field to categorize request duration into buckets. We can't easily query > in Kibana in
+	// OpenShift deployments due to logging limitations, so these constant buckets give us a small number of
+	// constant search strings we can use to identify slow reconcile loops.
+	for _, millis := range elapsedDurationBuckets {
+		if dur.Milliseconds() >= millis {
+			fields["elapsedMillisGT"] = millis
+			break
+		}
+	}
```

Comment:
I think we can use https://golang.org/pkg/sort/#SearchInts maybe because that uses binary search instead..
for this sort array i know this won't be a lot better just saying..